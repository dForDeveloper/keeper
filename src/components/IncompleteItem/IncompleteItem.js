import React from 'react';
import PropTypes from 'prop-types';
import deleteicon from '../../images/deleteicon.svg';
import uncheckedicon from '../../images/uncheckedicon.svg';

const IncompleteItem = (props) => {
  const { listItems, id, description, focusedListItemID } = props;
  return (
    <span key={id} className='NoteForm--span--incomplete'>
      <img
        src={uncheckedicon}
        className='NoteForm--icon--unchecked'
        onClick={() => props.handleComplete(listItems, id)}
        alt='unchecked icon'
      />
      <input
        name={id}
        value={description}
        autoFocus={id === focusedListItemID}
        onChange={props.handleChange}
        className='NoteForm--list-item'
      />
      <img
        src={deleteicon}
        className='NoteForm--icon--delete'
        onClick={() => props.handleItemDelete(listItems, id)}
        alt='delete icon'
      />
    </span>
  )
}

IncompleteItem.propTypes = {
  listItems: PropTypes.array,
  id: PropTypes.string,
  description: PropTypes.string,
  focusedListItemID: PropTypes.string,
  handleComplete: PropTypes.func,
  handleChange: PropTypes.func,
  handleItemDelete: PropTypes.func
}

export default IncompleteItem;