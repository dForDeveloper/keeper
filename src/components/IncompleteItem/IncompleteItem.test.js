import React from 'react';
import { shallow } from 'enzyme';
import IncompleteItem from './IncompleteItem';

describe('IncompleteItem', () => {
  let wrapper;
  const mockProps = {
    id: 'abc',
    description: 'do stuff',
    focusedListItemID: 'abc',
    handleComplete: jest.fn(),
    handleChange: jest.fn(),
    handleItemDelete: jest.fn()
  }
  beforeEach(() => {
    wrapper = shallow(<IncompleteItem {...mockProps} />);
  });

  it('should map the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleComplete when uncheckedicon is clicked', () => {
    wrapper.find('.NoteForm--icon--unchecked').simulate('click');
    expect(mockProps.handleComplete).toHaveBeenCalledWith('abc');
  });

  it('should call handleItemDelete when deleteicon is clicked', () => {
    wrapper.find('.NoteForm--icon--delete').simulate('click');
    expect(mockProps.handleItemDelete).toHaveBeenCalledWith('abc');
  });

  it('should call handleChange when the input is changed', () => {
    wrapper.find('.NoteForm--list-item').simulate('change');
    expect(mockProps.handleChange).toHaveBeenCalled();
  });
});