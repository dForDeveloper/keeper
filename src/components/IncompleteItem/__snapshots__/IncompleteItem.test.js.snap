// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncompleteItem should map the snapshot 1`] = `
<span
  className="NoteForm--span--incomplete"
>
  <img
    alt="unchecked icon"
    className="NoteForm--icon--unchecked"
    onClick={[Function]}
    src="uncheckedicon.svg"
  />
  <input
    autoFocus={true}
    className="NoteForm--list-item"
    name="abc"
    onChange={[MockFunction]}
    value="do stuff"
  />
  <img
    alt="delete icon"
    className="NoteForm--icon--delete"
    onClick={[Function]}
    src="deleteicon.svg"
  />
</span>
`;
